libV2/utils/filter.js

```diff
@@ -0,0 +1,54 @@
+const assert = require('assert');
+
+/**
+ * filterObject
+ *
+ * Constructs a function meant for filtering Objects by the value of a key
+ * Returned function returns a boolean with false meaning the object was present
+ * in the filter allowing the function to be passed directly to Array.filter etc.
+ *
+ * @param {string} key - Object key to inspect
+ * @param {Object} filter
+ * @param {Set} [filter.allow] - Set containing keys to include
+ * @param {Set} [filter.deny] - Set containing keys to not include
+ * @returns {function(Object): bool}
+ */
+
+function filterObject(key, { allow, deny }) {
+    if (allow && deny) {
+        throw new Error('You can not define both an allow and a deny list.');
+    }
+    if (!allow && !deny) {
+        throw new Error('You must define either an allow or a deny list.');
+    }
+    if (allow) {
```

Comment:
Is it worth having an allow set, since AFAIK it was not part of the requirements? Ok to keep it but we need to be wary of the added complexity.

Reply:
I think, from a technical perspective, that implementing `allow` at the same time as `deny` makes sense as it allows us to design an initial solution that supports both rather than needing rework if the feature is eventually asked for. In my opinion the small amount of extra code and complexity needed to support `allow` is worth it.

Reply:
OK, good for me it makes sense.